doc/psidk/src/specifications/app-architecture/supervisor.md

```diff
@@ -20,48 +20,127 @@ Supervisor is responsible for:
+    params: { ... },
+})
+```
+
+#### Proposed Supervisor API
+
+```
+const services = [{
+    service: "sample_srvc_name",
+}, {
+    service: "srvc_name2",
+    plugin: "plugin2" // optional; assumed to be simply "plugin" by default
+}
+];
+
+// Supervisor.connect(...) will internally create a uuid to represent the client session
+const supervisor = await Supervisor.connect(options: {
+    services   // optional: for, e.g., preloading. the <wasm filename>.wasm
+    // if not provided, Supervisor will load the wasm specified in supervisor.call()
+});
+
+const res = await supervisor.call({
+    service: "service_name",
+    plugin: "plugin_name", // optional
+    method: "method_name",
+    params: {
+        arg1: "arg value 1"
+    }
+});
+```
+
+App1's call is routed through the Supervisor, which then routes the call to the intended app.
+This hierarchy of apps ensures all apps can verify the source/destination of their function calls is the Supervisor, which provides a security guaranty.
+
 ```svgbob
 .-------------------------------------------.
-| 🔒 app1.psibase                           |
+| 🔒 app1.psibase.tld                       |
 |-------------------------------------------|
 |                                           |
-|   .----------------------------------.    |
-|   | </> supervisor.psibase  [hidden] |    |
-|   |----------------------------------|    |
-|  <-->                                |    |
-|   |   .--------------------------.   |    |
-|   |   | </> app1.plugin.psibase  |   |    |
-|   |   |--------------------------|   |    |
-|   |  <-->                        |   |    |
-|   |   |                          |   |    |
-|   |   '--------------------------'   |    |
-|   |   .--------------------------.   |    |
-|   |   | </> app2.plugin.psibase  |   |    |
-|   |   |--------------------------|   |    |
-|   |  <-->                        |   |    |
-|   |   |                          |   |    |
-|   |   '--------------------------'   |    |
-|   '----------------------------------'    |
+|   .-------------------------------------. |
+|   | supervisor.psibase.tld [hidden]     | |
+|  <-->                                   | |
+|   |   .-----------------------------.   | |
+|   |   | app1.psibase.tld/plugin     |   | |
+|   |  <-->                           |   | |
+|   |   |                             |   | |
+|   |   '-----------------------------'   | |
+|   |   .-----------------------------.   | |
+|   |   | app2.psibase.tld/plugin     |   | |
+|   |  <-->                           |   | |
+|   |   |                             |   | |
+|   |   '-----------------------------'   | |
+|   '-------------------------------------' |
 |                                           |
 '-------------------------------------------'
 ```
 
+When App1 instantiates the Supervisor, Supervisor is embedded in a hidden iframe.
+The Supervisor, in turn, embeds another iframe that contains `app2`'s plugin.
```

Comment:
I would delete the paragraph under the new diagram because it's redundant now given what you wrote here